core/blockchain.go

```diff
@@ -1190,15 +1193,16 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, []
 		}
```

Comment:
Can we also do `events = append(events, ChainEvent{block, block.Hash(), nil})` for prefix known blocks?

Reply:
Yeah, my reluctance around these is that I'm not sure if we're supposed to fire canon or side things? Are we sure it's **always** canon? If so we can definitely do.

But how do we fire logs?

Reply:
In this case, known blocks are canonical since the `externTd > localTd`

Reply:
I'm not really sure we should announce: if we get 10 new blocks from the network, and we in the mean time imported the first 2, we should not double announce the chain events.